src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final String LINE_SEPARATOR = "\n";
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)  {
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo).append(LINE_SEPARATOR);
+
+        LocalDate from = LocalDate.now();
+        LocalDate to = LocalDate.now();
+
+        try {
+            from = LocalDate.parse(dateFrom, formatter);
+            to = LocalDate.parse(dateTo, formatter);
+
+        } catch (DateTimeParseException e) {
+            System.out.println("Error parsing date " + e);
```

Comment:
The error message 'Error parsing date' is too generic. Consider providing more context, such as which date failed to parse, to make debugging easier.